### PR TITLE
deep linking aliases

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

**Comment by six7** [comment]
Can you add tests for this scenario? The alias files should already have some tests setup, let's extend them.

**Comment by six7** [comment]
fyi: theres some failing tests @swordEdge

**Comment by swordEdge** [comment]
> Can you add tests for this scenario? The alias files should already have some tests setup, let's extend them.

done!

**Comment by swordEdge** [comment]
> * Don't show the red dot

fixed!
> * ideally, it would show the fully resolved value, not $16. In that case, 16px.

fixed!

File: src/utils/alias/getAliasValue.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { convertToRgb } from '../color';
 import { findReferences } from '../findReferences';
 import { isSingleTokenValueObject } from '../is';
 import { checkAndEvaluateMath } from '../math';
+import { theme } from '@/stitches.config';
```
Comment (six7) [inline src/utils/alias/getAliasValue.ts:6]: i think this is a mistake? the stitches config is not required here

File: src/utils/alias/getAliasValue.ts [hunk 2]
```diff
@@ -38,7 +39,11 @@ export function getAliasValue(token: SingleToken | string | number, tokens: Sing
       tokenReferences.forEach((reference, index) => {
         const resolvedReference = resolvedReferences[index];
         const stringValue = String(resolvedReference);
-        const resolved = checkAndEvaluateMath(stringValue);
+        let resolved = checkAndEvaluateMath(stringValue);
+        if (String(resolved).startsWith("$")) {
+          const sizeProperty = String(resolved).slice(1, String(resolved).length);
+            resolved = theme.space[`${sizeProperty}`].value;
```
Comment (six7) [inline src/utils/alias/getAliasValue.ts:45]: why is this here? our stitches theme config is definitely not something our users would want to alias?

**Review by six7 — CHANGES_REQUESTED** [review]
i think there's some unintended changes here, let's not include our stitches theme config in user's aliases

i also noticed it currently still shows the red dot if the user used the $ notation, the preview doesnt work as well if the user used the $ notation.

image
image
image
image

File: src/utils/alias/getAliasValue.ts (hunk 1 above)
Comment (swordEdge) [reply src/utils/alias/getAliasValue.ts]: Fixed!

File: src/utils/alias/getAliasValue.ts (hunk 2 above)
Comment (swordEdge) [reply src/utils/alias/getAliasValue.ts]: Fixed!

**Comment by swordEdge** [comment]
> i think there's some unintended changes here, let's not include our stitches theme config in user's aliases

Fixed!

**Comment by swordEdge** [comment]
> Can you add tests for this scenario? The alias files should already have some tests setup, let's extend them.

Done!

File: src/utils/alias/getAliasValue.ts [hunk 3]
```diff
@@ -9,22 +9,58 @@ export function getAliasValue(token: SingleToken | string | number, tokens: Sing
...
+          else 
+            nameToLookFor = ref.substring(1);
+
+          if (
+            (typeof token === 'object' && nameToLookFor === token.name) || 
+            nameToLookFor === token
+          ) 
+            return null;
+
+          type TokenNameNodeType = string | undefined;
```
Comment (six7) [inline src/utils/alias/getAliasValue.ts:33]: This can probably move to the top of the file or to its own file?

**Review by six7 — APPROVED** [review]
Looking good! I had one comment regarding where that type is being declared. Other than that this works great!

**Review by LiamMartens — APPROVED** [review]

File: src/utils/alias/getAliasValue.ts (hunk 3 above)
Comment (swordEdge) [reply src/utils/alias/getAliasValue.ts:33]: Fixed!

**Pull request merged**